pkg/migration/converter.go

```diff
@@ -74,9 +74,29 @@ func sanitizeResource(m map[string]any) map[string]any {
 	if len(metadata) == 0 {
 		delete(m, "metadata")
 	}
+	removeNilValuedKeys(m)
 	return m
 }
 
+func removeNilValuedKeys(m map[string]interface{}) {
```

Comment:
As far as I understand, we are cleaning here by traversing the map. I guess this is something we do to avoid seeing nil values in the manifest.

Reply:
Yes, correct. Added a comment explaining the purpose.